addons/assigngear/functions/fnc_assignGear.sqf

```diff
@@ -106,42 +106,54 @@ _unit setUnitLoadout (configFile >> 'EmptyLoadout');
                 private _weapon = selectRandom _x;
                 if !(_weapon isEqualTo '') then {_unit addWeapon _weapon};
             };
-            case 14: { // scope
+            case 14: { // primarymagazine
+                private _magazine = selectRandom _x;
+                if !(_magazine isEqualTo '') then {_unit addPrimaryWeaponItem _magazine};
+            };
+            case 15: { // scope
```

Comment:
I'll make a separate PR to pull these indexes out to make these kinds of reactors easier in the future. Similar to #300

Reply:
Going to do this in this PR actually since I need to define an order to the `loadoutArray` to guarantee that weapons are added before magazines.